crates/bevy_render/src/camera/camera.rs

```diff
@@ -22,38 +22,144 @@ use bevy_transform::components::GlobalTransform;
 use bevy_utils::HashSet;
 use bevy_window::{WindowCreated, WindowId, WindowResized, Windows};
 use serde::{Deserialize, Serialize};
-use std::borrow::Cow;
+use std::{borrow::Cow, ops::Range};
 use wgpu::Extent3d;
 
+/// Render viewport configuration for the [`Camera`] component.
+///
+/// The viewport defines the area on the screen that the camera renders its image to.
+/// You can overlay multiple cameras in a single window using viewports to create effects like like split screen, minimaps, portals and character viewers.
+// TODO: remove reflect_value when possible
+#[derive(Reflect, Debug, Clone, Serialize, Deserialize)]
+#[reflect_value(Default, Serialize, Deserialize)]
+pub struct Viewport {
+    /// The physical position to render this viewport to within the [`RenderTarget`] of this [`Camera`].
+    /// (0,0) corresponds to the top-left corner
+    pub physical_position: UVec2,
+    /// The physical size of the viewport rectangle to render to within the [`RenderTarget`] of this [`Camera`].
+    /// The origin of the rectangle is in the top-left corner.
+    pub physical_size: UVec2,
+    /// The minimum and maximum depth to render to (on a scale from 0.0 to 1.0).
+    pub depth: Range<f32>,
+}
+
+impl Default for Viewport {
+    fn default() -> Self {
+        Self {
+            physical_position: Default::default(),
+            physical_size: Default::default(),
+            depth: 0.0..1.0,
+        }
+    }
+}
+
+/// Information about the current [`RenderTarget`].
+#[derive(Default, Debug, Clone)]
+pub struct RenderTargetInfo {
+    /// The physical size of this render target (ignores scale factor).
+    pub physical_size: UVec2,
+    /// The scale factor of this render target.
+    pub scale_factor: f64,
```

Comment:
Why `f64` here when we use `f32` basically everywhere else?

Reply:
Winit uses f64 for the scale factor. Thats really the only reason.